tiledb/sm/group/group.cc

```diff
@@ -300,6 +308,47 @@ Status Group::get_query_type(QueryType* query_type) const {
   return Status::Ok();
 }
 
+void Group::delete_group(const URI& uri, bool recursive) {
+  // Check that group is open
+  if (!is_open_) {
+    throw DeleteGroupStatusException("Group is not open");
+  }
+
+  // Check that query type is MODIFY_EXCLUSIVE
+  if (query_type_ != QueryType::MODIFY_EXCLUSIVE) {
+    throw DeleteGroupStatusException("Query type must be MODIFY_EXCLUSIVE");
+  }
+
```

Comment:
Let's add a comment here as to what this block of code does. Same above if (remote_) { below.